app/models/concerns/host_info.rb

```diff
@@ -0,0 +1,56 @@
+# This module provides a method that exposes parameter information for a given host.
+# The hash is extensible, so any plugin/module can add a provider that will add its
+# information to the method.
+# Usage:
+# host = Host::Managed.find(my_host_id)
+# host.info
+# => A hash of hashes that describes all parameters associated with the host.
+#
+# Writing an extension:
+# Create an ActiveSupport::Concern with a method that will return a hash with the
+# info that should be added and register the method by calling register_info_provider
+# module MyInfoProvider
+#   extend ActiveSupport::Concern
+#
+#   def my_info
+#     { 'my_prop1' => 'my_val' }
+#   end
+#
+#   included do
+#     register_info_provider :my_info
+#   end
```

Comment:
Could you please document this example to [redmine wiki](http://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin) when this is merged? Also could you extend [Plugin DSL](https://github.com/theforeman/foreman/blob/12612809c5deb885b13ceaf36b6a147c21688eb5/app/services/foreman/plugin.rb#L273) so plugin authors can register their info providers easily from initializer?